tests/unit/type/hash-field-expire.tcl

```diff
@@ -1261,3 +1269,237 @@ start_server {tags {"external:skip needs:debug"}} {
         r config set hash-max-listpack-value 64
     }
 }
+
+start_server {tags {"external:skip needs:debug"}} {
+    foreach type {listpack ht} {
+        if {$type eq "ht"} {
+            r config set hash-max-listpack-entries 0
+        } else {
+            r config set hash-max-listpack-entries 512
+        }
+
+        test "Command rewrite and expired hash fields are propagated to replica ($type)" {
+            start_server {overrides {appendonly {yes} appendfsync always} tags {external:skip}} {
+
+                set aof [get_last_incr_aof_path r]
+                r hset h1 f1 v1 f2 v2
+
+                assert_equal [r hpexpire h1 10 FIELDS 1 f1] [list  $E_OK]
+
+                r hpexpire h1 11 FIELDS 1 f2
```

Comment:
11 ms is too short, i get a test failure here.
the hpexpire* commands are all the same.

```
Command rewrite and expired hash fields are propagated to replica (ht) in tests/unit/type/hash-field-expire.tcl
Expected 'hdel h1 f1' to match 'hset h2 f1 v1 f2 v2 f3 v3 f4 v4' (context: type source line 971 file /home/sundb/data/redis_fork/tests/support/util.tcl cmd {assert_match [lindex $patterns $j] [read_from_aof $fp]} proc ::assert_aof_content level 1)
```